service/service.go

```diff
@@ -155,6 +161,10 @@ func (srv *Service) Start(ctx context.Context) error {
 		return err
 	}
 
+	if err := srv.host.serviceExtensions.NotifyConfigResolved(ctx, srv.collectorConf); err != nil {
+		return err
+	}
```

Comment:
Can `NotifyConfigResolved` be called right after the extensions are started? Or does it need to wait for pipelines to start?

Reply:
My goal originally was to ensure that the pipeline successfully starts before notifying the extension of the config. Thinking on it, we should record the config even for failure cases, so I will update this to notify extensions as soon as they are started.